infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogReplicationSinkManager.java

```diff
@@ -574,8 +570,10 @@ public void reset() {
 
     public void shutdown() {
         this.runtime.shutdown();
-        this.applyExecutor.shutdownNow();
-        isSinkManagerShutdown = true;
+    }
+
+    public static void shutdownApplyExecutor() {
```

Comment:
yes. The apply executor is changed to be shared by all the SinkManagers now. And this would only be shutdown when LR has to shut down.
So this method would now be called from SessionManager in the LR shutdown workflow